### PR TITLE
post: sign your code

### PR DESCRIPTION
**Pull request by navarroaxel**

### CODE REVIEW

File: posts/navarroaxel/01.sign-your-code.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+---
+title: Sign your code
+published: false
+description: You can sign your code in Git using a GPG key.
+tags: 'git, github, devops, security'
+---
+
+Did you know you can sign your code in Git using a GPG key? A lot of programmers don't know they can sign their Git commits using a signature created by themself.
```
Comment (pabloromeo) [inline posts/navarroaxel/01.sign-your-code.md:8]:
```suggestion
Did you know you can sign your code in Git using a GPG key? A lot of programmers don't know they can sign their Git commits using a signature created by themselves.
```

File: posts/navarroaxel/01.sign-your-code.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+This is fully integrated with Git and you can automatically sign your code with it.
+
+You can create a GPG key following [this guide](https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key) in GitHub, remember that you should register it in [GitHub](https://github.com/settings/keys), or [GitLab](https://gitlab.com/-/profile/gpg_keys).
+
+## Should I sign every commit?
+
+There is a discussion about this, because Linus Torvalds says that when you're signing a Git tag 🏷️, you're validating all the commits in the release, signed or not. If you **automatically** sign every commit in the repository, the source of the signature loses its sense.
+
+In the other hand, some Linux distros, like Arch Linux, uses Arch User (Git) Repositories made by users to build non-official supported packages. I mean, if you're compiling and installing a package via another user's instructions maybe you like the idea of the authors signing their work. ⚠️
```
Comment (pabloromeo) [inline posts/navarroaxel/01.sign-your-code.md:26]:
```suggestion
On the other hand, some Linux distros, like Arch Linux, uses Arch User (Git) Repositories made by users to build non-official supported packages. I mean, if you're compiling and installing a package via another user's instructions maybe you like the idea of the authors signing their work. ⚠️
```

**Review by pabloromeo — CHANGES_REQUESTED** [review]

**Review by pabloromeo — DISMISSED** [review]
If you want, you could add an image showing the green checkmarks from signed commits.

File: posts/navarroaxel/03.sign-your-code.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+You can use any name or email when you create a Git commit, but you can't sign a commit with a GPG key that doesn't belong to you.
+
+### What is GPG?
+
+The GNU Privacy Guard is an implementation of the OpenPGP standard and allows you to encrypt and sign your data and communications.
+
+This is fully integrated with Git and you can automatically sign your code with it.
+
+You can create a GPG key following [this guide](https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key) in GitHub, remember that you should register it in [GitHub](https://github.com/settings/keys), or [GitLab](https://gitlab.com/-/profile/gpg_keys).
```
Comment (pabloromeo) [inline posts/navarroaxel/03.sign-your-code.md:22]:
```suggestion
You can create a GPG key following [this guide](https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key) in GitHub. Remember that you should register it in [GitHub](https://github.com/settings/keys), or [GitLab](https://gitlab.com/-/profile/gpg_keys).
```

**Review by pabloromeo — CHANGES_REQUESTED** [review]

**Review by pabloromeo — APPROVED** [review]

**Review by EzequielMontel — APPROVED** [review]

**Pull request merged**